scripts/mainmenu_mapselector.js

```diff
@@ -513,16 +515,14 @@ class MapSelection {
 	static onNStateBtnChanged(panelID, state) {
 		const panel = $.GetContextPanel().FindChildTraverse(panelID);
 
-		for (let type of Array(3).keys()) {
-			panel.SetHasClass(MapSelNStateClasses[type], state === type);
-		}
+		for (let type of Array(3).keys()) panel.SetHasClass(MapSelNStateClasses[type], state === type);
 	}
 
 	/**
 	 * Only show the clear button if the search is not empty
 	 */
 	static onSearchChanged() {
-		MapSelection.searchClearButton.SetHasClass('search__clearbutton--hidden', this.searchTextEntry.text === '');
+		this.panels.searchText.SetHasClass('search__clearbutton--hidden', this.searchText.text === '');
```

Comment:
The second `this.searchText` needs to be `this.panels.searchText`?

Reply:
Good catch